code/ui/blocks/src/controls/Color.tsx

```diff
@@ -318,10 +325,13 @@ export const ColorControl: FC<ColorControlProps> = ({
   const { presets, addPreset } = usePresets(presetColors, color, colorSpace);
   const Picker = ColorPicker[colorSpace];
 
+  const readonly = argType?.table?.readonly;
```

Comment:
did you want the values to be `true|false` or `true|false|undefined`?
```suggestion
  const readonly = !!argType?.table?.readonly;
```